pkg/controller/kieapp/deploy_ui.go

```diff
@@ -159,24 +159,15 @@ func updateCSVlinks(reconciler *Reconciler, route *routev1.Route, operator *apps
 				log.Error("Failed to get CSV. ", err)
 				return
 			}
-			update := true
-			exists := false
 			url := fmt.Sprintf("https://%s", found.Spec.Host)
-			for i, link := range csv.Spec.Links {
-				if link.Name == constants.ConsoleLinkName {
-					exists = true
-					if link.URL == url {
-						update = false
-					} else {
-						csv.Spec.Links[i].URL = url
-					}
+			link := getConsoleLink(csv)
+			if link == nil || link.URL != url {
+				if link == nil {
+					csv.Spec.Links = append([]operatorsv1alpha1.AppLink{{Name: constants.ConsoleLinkName, URL: url}}, csv.Spec.Links...)
+				} else {
+					link.URL = url
```

Comment:
How does this get into the updated csv object? I'm not sure the refactor of this will work as you lose the necessary slice index necessary to make this change in csv object.

Reply:
If you mean the `else` part, it's a reference to the array element, so updating it should update the existing CSV object. Because the object was passed by reference and the element returned by reference, the update should work.